### PR TITLE
Further redistribution of code into modules

### PR DESCRIPTION
**Pull request by djc**
More suggestions for evening out module sizes and strengthening abstractions.

### CODE REVIEW

File: quinn-proto/src/connection.rs [hunk 1]
```diff
@@ -2743,15 +2741,11 @@ impl Connection {
         self.state.is_drained()
     }
 
-    pub fn accepted_0rtt(&self) -> bool {
```
Comment (Ralith) [inline quinn-proto/src/connection.rs]: This is a necessary part of the public API for 0-RTT support: higher-level code needs it to determine whether 0-RTT streams still exist following connection establishment.

File: quinn-proto/src/connection.rs [hunk 2]
```diff
@@ -2731,10 +2733,6 @@ impl Connection {
         });
     }
 
-    pub fn is_handshaking(&self) -> bool {
```
Comment (Ralith) [inline quinn-proto/src/connection.rs]: Similarly, this is convenient to determine whether 0-RTT acceptance is defined yet. You could reconstruct that information from events, but it's a lot easier to be able to just look it up.

File: quinn-proto/src/connection.rs (hunk 1 above)
Comment (djc) [reply quinn-proto/src/connection.rs]: Brought it back!

File: quinn-proto/src/connection.rs (hunk 2 above)
Comment (djc) [reply quinn-proto/src/connection.rs]: Brought it back.

**Pull request merged**